src/lib/mina_ledger/root.ml

```diff
@@ -60,19 +64,47 @@ module Make
                             and type converting_ledger = Unstable_db.t) =
 struct
   module Config = struct
-    type t = string
-
-    type backing_type = Stable_db
-
-    let exists_backing = Sys.file_exists
-
-    let with_directory ~backing_type:Stable_db ~directory_name = directory_name
-
-    let delete_any_backing = Mina_stdlib_unix.File_system.rmrf
-
-    let move_backing_exn ~src ~dst = Sys.rename src dst
-
-    let primary_directory = Fn.id
+    type backing_type = Stable_db | Converting_db [@@deriving equal]
+
+    type t = { top_directory : string; backing_type : backing_type }
```

Comment:
The `Config` module in `converting_merkle_tree.ml` is set up so that if the primary database is in the directory `primary_directory`, then the converting database will be in `primary_directory_converting`. The advantage of having the databases side by side like that is that the expected primary database directory won't change depending on the value of the `--hardfork-mode` flag.

So the config type here could be something like

```ocaml
type t = Stable_db_config of string
       | Converting_db_config of Converting_ledger.Config.t
```

and then `Converting_ledger.Config.with_primary` could be used to construct the actual `Converting_ledger.Config.t`.

(The names of these methods are not the most consistent).

Reply:
I personally would like approach in this PR instead, putting everything in one folder so operations like moving the root is easier to implement. WDYT?